plaso/parsers/dsv_parser.py

```diff
@@ -134,35 +195,28 @@ def ParseFileObject(self, parser_mediator, file_object):
     Raises:
       UnableToParseFile: when the file cannot be parsed.
     """
-    file_size = file_object.get_size()
-    # The csv module can consume a lot of memory, 1 GiB for a 100 MiB file.
-    # Hence that the maximum supported file size is restricted.
-    if file_size > self._MAXIMUM_SUPPORTED_FILE_SIZE:
-      display_name = parser_mediator.GetDisplayName()
-      raise errors.UnableToParseFile((
-          '[{0:s}] Unable to parse DSV file: {1:s} size of file exceeds '
-          'maximum supported size').format(self.NAME, display_name))
-
     # TODO: Replace this with detection of the file encoding via byte-order
     # marks. Also see: https://github.com/log2timeline/plaso/issues/1971
     if not self._encoding:
       self._encoding = parser_mediator.codepage
 
-    # The Python 2 csv module reads bytes and the Python 3 csv module Unicode
-    # reads strings.
-    if py2to3.PY_3:
-      line_reader = text_file.TextFile(file_object, encoding=self._encoding)
-    else:
-      line_reader = line_reader_file.BinaryLineReader(file_object)
-
-    # If we specifically define a number of lines we should skip, do that here.
-    for _ in range(0, self.NUMBER_OF_HEADER_LINES):
-      line_reader.readline()
-
-    reader = self._CreateDictReader(line_reader)
+    try:
+      if not self._HasExpectedLineLength(file_object):
```

Comment:
What if the file starts with 30 good lines and then is followed by a large blob of non-textual data?

Do we want to check for this edge case? Don't think it is blocking for this CL but let's add a todo or issue?

Reply:
I didn't think we needed to do this, and then I couldn't explain it adequately. Then I investigated a bunch more and found another bug in line_reader_file. It was doing unbounded reads, which was probably causing the memory issues in the first place. 

In the end though, we don't need to do anything for this edge case - there'll be a crash in the csv reader now, which is what we want.